### PR TITLE
Address some deprecation warnings

### PR DESCRIPTION
**Pull request by louwie17**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This PR address 3 different deprecation warnings:
- The use of the conditional block
- Using a very old version of React context ( which wasn't being used anymore anyhow )
- Remove use of position in `Dropdown` usage.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

**NOTE:** there are still a couple other warnings present, one is for the `tooltip` usage, but that only got deprecated in 6.4, so we may have to wait with fixing that until `6.5` is out.

**Testing link update**
1. Load this branch and enable the new product editor in **WooCommerce > Settings > Advanced > Features**
2. Open the browser console
3. Go to **Products > Add new**
4. Click the **Pricing** link in the list price
5. It should navigate to the price tab correctly without errors
6. Check your console, there should not be a warning present saying: `Legacy context API has been detected within a strict-mode tree`

**Testing conditional block**
1. Load this branch and enable the new product editor in **WooCommerce > Settings > Advanced > Features**
2. Open the browser console
3. Go to **Products > Add new**
4. Click the **Inventory** tab
5. **Track stock...** should be disabled
6. The **Stock status** and under advanced the **Restrictions** field should be present
7. Enable **Track stock...**
8. The **Available quantity** field should show and the **Stock status** should dissapear.
9. Under advanced the **When out of stock** and **email me when....** should should be present.
10. It should navigate to the price tab correctly without errors
11. Check your console, there should not be a warning present saying: `'woocommerce/conditional' block is deprecated. Please use 'hideConditions' attribute on any block instead.`

**Testing dropdown usage**
1. Load this branch and enable the new product editor in **WooCommerce > Settings > Advanced > Features**
2. Open the browser console
3. Go to **Products > Add new**
4. Click the Downloads dropdown and use the **Insert from URL** ( should work correctly )
6. Create a product with variations and use the dropdown filters in the variation table, they should work correctly.
8. Check your console, there should not be a warning present saying: `'position' prop in wp.components.Dropdown is deprecated since version 6.2. Please use 'popoverProps.placement' prop instead. Note: Note that the 'position' prop will override any values passed through the 'popoverProps.placement' prop.`
12. Also try the dropdowns usage in Analytics > Products filtering by date or **Analytics > Settings** using the default date range dropdown.
13. Check if the above warning from step 6 isn't shown.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: packages/js/components/src/advanced-filters/index.js [hunk 1]
```diff
@@ -320,7 +320,9 @@ class AdvancedFilters extends Component {
 						<div className="woocommerce-filters-advanced__add-filter">
 							<Dropdown
 								className="woocommerce-filters-advanced__add-filter-dropdown"
-								position="bottom center"
+								popoverProps={ {
+									position: 'bottom center',
+								} }
```
Comment (retrofox) [inline packages/js/components/src/advanced-filters/index.js]: > Note: use the placement prop instead when possible.
> [position prop](https://developer.wordpress.org/block-editor/reference-guides/components/popover/#position-yaxis-xaxis-optionalcorner)

Something like the following (not tested):
```suggestion
								popoverProps={ {
									placement: 'bottom',
								} }
```

File: packages/js/components/src/calendar/date-picker.js [hunk 2]
```diff
@@ -86,9 +86,12 @@ class DatePicker extends Component {
 			popoverProps = { inline: true },
 		} = this.props;
 
+		if ( ! popoverProps.position ) {
+			popoverProps.position = 'bottom center';
+		}
```
Comment (retrofox) [inline packages/js/components/src/calendar/date-picker.js]: Similarely to the previous comment:

```suggestion
		if ( ! popoverProps.position ) {
			popoverProps.placement = 'bottom';
		}
```

File: packages/js/components/src/ellipsis-menu/index.tsx [hunk 3]
```diff
@@ -87,7 +87,10 @@ const EllipsisMenu = ( {
 		<div className={ classnames( className, 'woocommerce-ellipsis-menu' ) }>
 			<Dropdown
 				contentClassName="woocommerce-ellipsis-menu__popover"
-				position="bottom left"
+				// @ts-expect-error missing prop in types.
+				popoverProps={ {
+					position: 'bottom left',
```
Comment (retrofox) [inline packages/js/components/src/ellipsis-menu/index.tsx]: https://github.com/woocommerce/woocommerce/pull/41845/files#r1419213870

File: packages/js/components/src/filter-picker/index.js [hunk 4]
```diff
@@ -294,7 +294,9 @@ class FilterPicker extends Component {
 				) }
 				<Dropdown
 					contentClassName="woocommerce-filters-filter__content"
-					position="bottom"
+					popoverProps={ {
+						position: 'bottom',
+					} }
```
Comment (retrofox) [inline packages/js/components/src/filter-picker/index.js]: https://github.com/woocommerce/woocommerce/pull/41845/files#r1419213870

File: packages/js/components/src/summary/index.js [hunk 5]
```diff
@@ -49,7 +49,9 @@ const SummaryList = ( { children, isDropdownBreakpoint, label } ) => {
 	return (
 		<Dropdown
 			className="woocommerce-summary"
-			position="bottom"
+			popoverProps={ {
+				position: 'bottom',
+			} }
```
Comment (retrofox) [inline packages/js/components/src/summary/index.js]: https://github.com/woocommerce/woocommerce/pull/41845/files#r1419213870

File: packages/js/product-editor/src/blocks/product-fields/downloads/downloads-menu/downloads-menu.tsx [hunk 6]
```diff
@@ -20,7 +20,10 @@ export function DownloadsMenu( {
 }: DownloadsMenuProps ) {
 	return (
 		<Dropdown
-			position="bottom left"
+			// @ts-expect-error missing prop in types.
+			popoverProps={ {
+				position: 'bottom left',
+			} }
```
Comment (retrofox) [inline packages/js/product-editor/src/blocks/product-fields/downloads/downloads-menu/downloads-menu.tsx]: https://github.com/woocommerce/woocommerce/pull/41845/files#r1419213870

File: packages/js/product-editor/src/blocks/product-fields/downloads/insert-url-menu-item/insert-url-menu-item.tsx [hunk 7]
```diff
@@ -70,7 +70,10 @@ export function InsertUrlMenuItem( {
 
 	return (
 		<Dropdown
-			position="middle left"
+			// @ts-expect-error missing prop in types.
+			popoverProps={ {
+				position: 'middle left',
+			} }
```
Comment (retrofox) [inline packages/js/product-editor/src/blocks/product-fields/downloads/insert-url-menu-item/insert-url-menu-item.tsx]: https://github.com/woocommerce/woocommerce/pull/41845/files#r1419213870

**Comment by louwie17** [comment]
Good suggestion @retrofox, I updated all the uses of `position` to make use of `placement` instead.

File: packages/js/product-editor/src/components/variations-table/downloads-menu-item/downloads-menu-item.tsx [hunk 8]
```diff
@@ -139,7 +139,10 @@ export function DownloadsMenuItem( {
 
 	return (
 		<Dropdown
-			position="middle right"
+			// @ts-expect-error missing prop in types.
+			popoverProps={ {
+				placement: 'right-start',
```
Comment (retrofox) [inline packages/js/product-editor/src/components/variations-table/downloads-menu-item/downloads-menu-item.tsx:144]: I think `position="middle right"` is equivalent to `placement: 'right'`?

File: plugins/woocommerce-admin/client/customize-store/assembler-hub/sidebar/sidebar-navigation-screen-logo.tsx [hunk 9]
```diff
@@ -451,7 +451,8 @@ export const SidebarNavigationScreenLogo = () => {
 								popoverProps={ {
 									className:
 										'woocommerce-customize-store__logo-dropdown-popover',
-									position: 'bottom left',
+									// @ts-expect-error outdated TS.
+									placement: 'left',
```
Comment (retrofox) [inline plugins/woocommerce-admin/client/customize-store/assembler-hub/sidebar/sidebar-navigation-screen-logo.tsx]: should it be `left-end`?

File: packages/js/product-editor/src/components/variations-table/downloads-menu-item/downloads-menu-item.tsx (hunk 8 above)
Comment (louwie17) [reply packages/js/product-editor/src/components/variations-table/downloads-menu-item/downloads-menu-item.tsx:144]: Yeah, although I updated it to use `right-start` instead, it seemed better:
`right`:
Screenshot 2023-12-12 at 3 31 58 PM


`right-start`:
Screenshot 2023-12-12 at 3 32 08 PM

File: plugins/woocommerce-admin/client/customize-store/assembler-hub/sidebar/sidebar-navigation-screen-logo.tsx (hunk 9 above)
Comment (louwie17) [reply plugins/woocommerce-admin/client/customize-store/assembler-hub/sidebar/sidebar-navigation-screen-logo.tsx]: Actually not sure how I ended up with `left`, this should be `bottom-end`

Comment (louwie17) [reply plugins/woocommerce-admin/client/customize-store/assembler-hub/sidebar/sidebar-navigation-screen-logo.tsx]: Updated here: https://github.com/woocommerce/woocommerce/pull/41845/commits/9dce5272310c86c2477551315fd75c516e26b22c

**Review by retrofox — APPROVED** [review]
LGTM. Thanks for addressing these issues 💯

**Pull request merged**